service/src/main/kotlin/fi/espoo/evaka/incomestatement/IncomeStatementQueries.kt

```diff
@@ -484,25 +486,57 @@ data class IncomeStatementAwaitingHandler(
     val id: IncomeStatementId,
     val created: HelsinkiDateTime,
     val startDate: LocalDate,
+    val incomeEndDate: LocalDate?,
     val handlerNote: String,
     val type: IncomeStatementType,
     val personId: PersonId,
     val personName: String,
     val primaryCareArea: String?,
 )
 
-// language=SQL
-private const val awaitingHandlerQuery =
-    """
-SELECT DISTINCT ON (i.created, i.start_date, i.id)
+private fun awaitingHandlerQuery(
+    today: LocalDate,
+    areas: List<String>,
+    providerTypes: List<ProviderType>,
+    sentStartDate: LocalDate?,
+    sentEndDate: LocalDate?,
+    placementValidDate: LocalDate?,
+) = QuerySql {
+    val filters =
+        Predicate.all(
```

Comment:
Käytä `PredicateSql` tavallisen `Predicate` sijaan, koska tässä ehdot kohdistuu useampaan tauluun.

`PredicateSql` = mikä vaan SQL-predikaatti (palauttaa boolean, kelpaa WHEREen). Suoraan laitettavissa kyselyyn ilman `forTable`-kutsua.
`Predicate` = yhteen tauluun kohdistuva predikaatti, ja taulun nimi on parametrisoitu (`it`). Pitää sanoa `forTable` käyttöpaikassa jotta saadaan taulun nimi paikalleen, ja pellin alla tuo `forTable` palauttaa `PredicateSql`-arvon

Meillä on myös helpperit not null -filtteröinnille, eli helpointa on sanoa

```kotlin
PredicateSql.allNotNull(
  PredicateSql { ... }.takeIf { ... },
  PredicateSql { ... }.takeIf { ... },
)
```

Reply:
Hyvä tieto 👍 Korjasin tämän ja muutin samalla pari muutakin paikkaa käyttämään `PredicateSql`:ää.